File: api/src/main/java/net/blueshell/api/repository/TelemetryRepository.java
package net.blueshell.api.repository;

import jakarta.validation.constraints.NotNull;
import net.blueshell.api.base.BaseRepository;
import net.blueshell.api.model.Telemetry;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface TelemetryRepository extends BaseRepository<Telemetry, UUID> {

    @org.jetbrains.annotations.NotNull
    @Query("SELECT n FROM Telemetry n ORDER BY n.createdAt DESC")
    @Override
    Page<Telemetry> findAll(@org.jetbrains.annotations.NotNull @NotNull Pageable pageable);

    @org.jetbrains.annotations.NotNull
    @Query("SELECT n FROM Telemetry n ORDER BY n.createdAt DESC")
    @Override
    List<Telemetry> findAll();
}

File: api/src/main/kotlin/net/blueshell/api/repository/TelemetryRepository.kt
package net.blueshell.api.repository

import jakarta.validation.constraints.NotNull
import net.blueshell.api.base.BaseRepository
import net.blueshell.api.model.Telemetry
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface TelemetryRepository : BaseRepository<Telemetry, Long> {
    @Query("SELECT n FROM Telemetry n ORDER BY n.createdAt DESC")
    override fun findAll(@NotNull pageable: @NotNull Pageable): Page<Telemetry>

    @Query("SELECT n FROM Telemetry n ORDER BY n.createdAt DESC")
    override fun findAll(): MutableList<Telemetry>
}
